app/presenters/chf/dates_of_work_for_display.rb

```diff
@@ -0,0 +1,98 @@
+module CHF
+  class DatesOfWorkForDisplay
+
+    BEFORE = "before"
+    AFTER = "after"
+    CENTURY = "century"
+    CIRCA = "circa"
+    DECADE = "decade"
+    UNDATED = "Undated"
+
+    def initialize(dates_of_work)
+      @dates_for_display = dates_of_work.collect {|d| display_label(d)}
+    end
+
+    #Return an array of strings, each containing a formatted date.
+    def display_dates
+      @dates_for_display
+    end
+
+    def display_label(date_of_work)
+      start_string = qualified_date(date_of_work.start,
+        date_of_work.start_qualifier)
+      finish_string = qualified_date(date_of_work.finish,
+        date_of_work.finish_qualifier)
+
+      if finish_string.blank?
+        date_string = start_string
+      else
+        # careful, this is an en dash:
+        date_string = [start_string, finish_string].compact.join(" – ")
+      end
+
+      if date_of_work.note.present?
+        date_string << " (#{date_of_work.note})"
+      end
```

Comment:
Only noticed cause of the honeybadger error ha (no big deal! that's what staging is for). 

`<<` actually mutates the string. a non-mutating way to do it would be `date_string = date_string + whatever`, assuming you have a `date_string=` writer method. 

Mutating strings is generally considered undesirable unless you really want/need to do it for performance or whatever. 

Mutating strings passed in _by someone else_ is especially evil. 

Mutating strings passed in by someone else that are [frozen](http://blog.honeybadger.io/when-to-use-freeze-and-frozen-in-ruby/) will get you an error -- and thank whatever code froze it before passing it in, for doing so and preventing a much more mysterious error when one mutates someone elses string, heh.